### PR TITLE
Bugfix/avoid sync getting stuck

### PR DESCRIPTION
**Pull request by mgallien**

### CODE REVIEW

File: src/libsync/bulkpropagatorjob.cpp [hunk 1]
```diff
@@ -231,6 +231,26 @@ void BulkPropagatorJob::triggerUpload()
...
 }
 
+void BulkPropagatorJob::checkPropagationIsDone()
+{
+    if (_items.empty()) {
+        if (!_jobs.empty()) {
+            // just wait for the other job to finish.
+            return;
+        }
+        if (!_pendingChecksumFiles.empty()) {
```
Comment (allexzander) [inline src/libsync/bulkpropagatorjob.cpp:241]: @mgallien You could combine it with previous `if`:

`if (!_jobs.empty())` -> `if (!_jobs.empty() && !_pendingChecksumFiles.empty())`

Comment (mgallien) [reply src/libsync/bulkpropagatorjob.cpp:241]: sure

**Review by camilasan — APPROVED** [review]

**Review by allexzander — APPROVED** [review]

**Pull request merged**